### PR TITLE
Transfer - Make curProcessedUploadChunks local variable

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Tests - https://github.com/jfrog/jfrog-cli/pull/2386
Make the curProcessedUploadChunks global variable private in producerConsumerWrapper.

### CODE REVIEW

File: artifactory/commands/transferfiles/producerconsumer.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	// This Producer-Consumer is used to upload chunks, initialized in newProducerConsumerWrapper; each uploading thread waits to be given tasks from the queue.
+	chunkUploaderProducerConsumer parallel.Runner
+	// This Producer-Consumer is used to execute AQLs and build chunks from the AQLs' results. The chunks data is sent to the go routines that will upload them.
+	// Initialized in newProducerConsumerWrapper; each builder thread waits to be given tasks from the queue.
+	chunkBuilderProducerConsumer parallel.Runner
+	// Errors related to chunkUploaderProducerConsumer and chunkBuilderProducerConsumer are logged in this queue.
+	errorsQueue *clientUtils.ErrorsQueue
+	// This variable holds the total number of upload chunk that were sent to the source Artifactory instance to process.
+	// Together with this mutex, they control the load on the user plugin and couple it to the local number of threads.
+	curProcessedUploadChunks   int
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/producerconsumer.go:20]: curProcessedUploadChunks --> totalProcessedUploadChunks

File: artifactory/commands/transferfiles/producerconsumer.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	if producerConsumerWrapper.curProcessedUploadChunks < GetChunkUploaderThreads() {
+		producerConsumerWrapper.curProcessedUploadChunks++
+		return true
+	}
+	return false
+}
+
+// Reduces the current total number of upload chunks processed. Called when an upload chunks doesn't require polling for status -
+// if it's done processing, or an error occurred when sending it.
+func (producerConsumerWrapper *producerConsumerWrapper) reduceCurProcessedChunks() {
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/producerconsumer.go:38]: reduceCurProcessedChunks --> devProcessedChunks

File: artifactory/commands/transferfiles/producerconsumer.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	errorsQueue *clientUtils.ErrorsQueue
+	// This variable holds the total number of upload chunk that were sent to the source Artifactory instance to process.
+	// Together with this mutex, they control the load on the user plugin and couple it to the local number of threads.
+	curProcessedUploadChunks   int
+	processedUploadChunksMutex sync.Mutex
+}
+
+// Checks whether the total number of upload chunks sent is lower than the number of threads, and if so, increments it.
+// Returns true if the total number was indeed incremented.
+func (producerConsumerWrapper *producerConsumerWrapper) incrCurProcessedChunksWhenPossible() bool {
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/producerconsumer.go:26]: incrCurProcessedChunksWhenPossible --> incProcessedChunksWhenPossible

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: artifactory/commands/transferfiles/producerconsumer.go (hunk 2 above)
Comment (yahavi) [reply artifactory/commands/transferfiles/producerconsumer.go]: Changed to decProcessedChunks

**Pull request merged**